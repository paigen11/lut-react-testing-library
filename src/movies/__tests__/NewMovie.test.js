import React from 'react';
import { render, cleanup } from 'react-testing-library';
import NewMovie from '../NewMovie';

afterEach(cleanup);

test('<NewMovie />', () => {
  const { getByTestId, queryByTestId, container } = render(<NewMovie />);
  expect(getByTestId('page-title').textContent).toBe('New Movie');
  expect(queryByTestId('movie-form')).toBeTruthy();

  // to update snapshots after changes made to component, type u in cmd line after tests run
  expect(container.firstChild).toMatchSnapshot();
});
