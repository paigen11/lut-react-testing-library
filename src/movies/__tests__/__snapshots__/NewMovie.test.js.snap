// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Movie
  </h1>
  <div>
    <form
      data-testid="movie-form"
    >
      <label
        for="text"
      >
        Text
        <input
          type="text"
          id="text"
        />
      </label>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
